.github/workflows/ci.yml

```diff
@@ -98,8 +99,8 @@ jobs:
         run: timeout 10s poetry run pip --version || rm -rf .venv
 
       - name: Install dependencies
-        if: steps.cache-pip.outputs.cache-hit != 'true'
-        run: poetry install
+        if: steps.cache-pip.outputs.cache-hit != 'true' || needs.diff.outputs.related == 'true'
+        run: poetry install -E "docs"
```

Comment:
Why don't we just cache all the dependencies we need for CI at once?

Reply:
yep we should do that instead.